pkg/uprobetracer/tracer.go

```diff
@@ -212,11 +215,16 @@ func (t *Tracer[Event]) attachUprobe(file *os.File) (link.Link, error) {
 		if err != nil {
 			return nil, fmt.Errorf("reading USDT metadata: %w", err)
 		}
-		return ex.Uprobe(t.attachSymbol, t.prog,
+		l, err := ex.Uprobe(t.attachSymbol, t.prog,
 			&link.UprobeOptions{
 				Address:      attachInfo.attachAddress,
 				RefCtrOffset: attachInfo.semaphoreAddress,
+				Cookie:       realInodePtr,
```

Comment:
cilium/ebpf supports two Linux APIs to add a uprobe:
1. Old API: tracefs
2. New API: PMU (using `unix.PerfEventOpen`), available since Linux 4.17 (https://github.com/torvalds/linux/commit/33ea4b24277b06dbc55d7f5772a46f029600255e)

The attach cookie (https://github.com/torvalds/linux/commit/82e6b1eee6a8875ef4eacfd60711cce6965c6b04) is only supported with the new API (Linux >=4.17).

But this is fine because the program with `bpf_get_attach_cookie` will be rejected in Linux < 5.15, so it won't be a silent failure.